front/app/components/IdeaCards/shared/Filters/StatusFilterDropdown.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useMemo } from 'react';
+
+import { capitalize } from 'lodash-es';
+
+import useIdeaStatuses from 'api/idea_statuses/useIdeaStatuses';
+
+import useLocalize from 'hooks/useLocalize';
+
+import FilterSelector from 'components/FilterSelector';
+
+import { FormattedMessage, useIntl } from 'utils/cl-intl';
+
+import messages from '../../messages';
+
+interface Props {
+  selectedStatusIds: string[];
+  alignment: 'left' | 'right';
+  onChange: (value: string[]) => void;
+  participationMethod: 'ideation' | 'proposals';
+}
+
+const StatusFilterDropdown = ({
+  selectedStatusIds,
+  alignment,
+  onChange,
+  participationMethod,
+}: Props) => {
+  const localize = useLocalize();
+  const { formatMessage } = useIntl();
+
+  const { data: ideaStatuses } = useIdeaStatuses({
+    participation_method: participationMethod,
+  });
+
+  const options = useMemo(() => {
+    return ideaStatuses
+      ? [
+          {
+            text: formatMessage(messages.allStatuses),
+            value: '',
+          },
+          ...ideaStatuses.data.map((status) => ({
+            text: capitalize(localize(status.attributes.title_multiloc)),
+            value: status.id,
+          })),
+        ]
+      : [];
+  }, [ideaStatuses, localize, formatMessage]);
+
+  if (!ideaStatuses || ideaStatuses.data.length === 0) {
+    return null;
+  }
+
+  return (
+    <FilterSelector
+      id="e2e-idea-filter-selector"
+      title={
+        <FormattedMessage
+          {...messages.statusesTitle}
+          key={`status-title-${Math.floor(Math.random() * 100000000)}`}
```

Comment:
Why is the random necessary? And if it is, isn't there something more deterministic we can use?

Reply:
Yep, weird.. Def not needed here, a copy-paste slip up.